### ISSUE TITLE
[Bug] Hitomi.la: Certain titles cause Tachiyomi to crash when searching

### ISSUE THREAD

**Issue by datreesezcup** [tachiyomiorg/tachiyomi-extensions#7224]
## Device information
- Tachiyomi version: Stable 0.10.12
- Android version: 9 (Pie)

## Source information
- Name: Hitomi.la
- Extension version: 1.2.5

## Steps to reproduce
Either via Global Search (with Hitomi.la enabled), or by searching within Hitomi.la, try to search for any of the following titles:
- 100 Nichigo ni S○X suru Tsuinte-chan
- A Girl Who Is Very Well-Informed About Weird Knowledge, Takayukashiki Souko-san
- a Story About a Totally Straightforward Girlfriend

(there are likely other titles that trigger this, this is what I've found so far)

### Expected behavior
Tachiyomi should search for this title in Hitomi.La, and either indicate that no manga were found, or return whatever manga it found to match the title.

### Actual behavior
Tachiyomi app crashes

## Other details
(I narrowed this issue down to Hitomi.la by searching titles using only one active source at a time)

While this issue causes Tachiyomi to crash on my physical Android 9 device (Tachiyomi Stable 0.10.12), it does *not* crash on my android emulator (Android 11, Tachiyomi Preview r3232 (98491d80)). Instead, tachiyomi indicates the basic error message on the screen, and the actual cause of the exception is logged.

**EDIT:
TL;DR: The exception that is generated from the emulator is `java.lang.NullPointerException: it must not be null`, and the stack trace indicates the exception is thrown from [Line 178 of Hitomi.kt](https://github.com/tachiyomiorg/tachiyomi-extensions/blob/6aac2a05397a8a1d381a31d842f64759cfdcb673/src/all/hitomi/src/eu/kanade/tachiyomi/extension/all/hitomi/Hitomi.kt#L178)**

Here is a text file of the stack trace generated on the Android emulator:
[android_emulator_stack.txt](https://github.com/tachiyomiorg/tachiyomi-extensions/files/6541452/android_emulator_stack.txt)

And if it helps, here is the crash log generated on my Android 9 device:
[tachiyomi_crash_logs.txt](https://github.com/tachiyomiorg/tachiyomi-extensions/files/6541463/tachiyomi_crash_logs.txt)

**Comment by arkon** [comment]
Hitomi.la is known to cause memory issues because the search is extremely complex and happens on the device.

Not much for us to do other than tell people to either download more RAM, use a different source, or try to search in your browser first.

**Comment by datreesezcup** [comment]
It doesn't seem to be an out-of-memory issue, it seems to be a programming issue (I think)

(Totally my bad, I should have *at least* put what the error message was from the emulator when I opened the issue. I'll edit it now.)

Specifically, the exception generated is:
`java.lang.NullPointerException: it must not be null`
and happens on [Line 178 of Hitomi.kt](https://github.com/tachiyomiorg/tachiyomi-extensions/blob/6aac2a05397a8a1d381a31d842f64759cfdcb673/src/all/hitomi/src/eu/kanade/tachiyomi/extension/all/hitomi/Hitomi.kt#L178)

Please correct me if I'm wrong. I suppose that somewhere along the line, a memory issue could screw up the search request, (But then wouldn't the exception thrown be an OutOfMemoryError, or something similar?)

**Comment by arkon** [comment]
My bad, I didn't look very closely.
